homeassistant/components/switch/mochad.py

```diff
@@ -59,17 +64,35 @@ def name(self):
 
     def turn_on(self, **kwargs):
         """Turn the switch on."""
-        self._state = True
         with mochad.REQ_LOCK:
-            self.device.send_cmd('on')
-            self._controller.read_data()
+            try:
+                self._state = True
```

Comment:
Put this at the end so that the state is not changed if there's an error in the communication with the device.

Reply:
Ok. Fixed in commit 9fbade8e7